packages/firestore/src/core/query.ts

```diff
@@ -643,8 +645,40 @@ export class FieldFilter extends Filter {
     }
   }
 
+  private static createKeyFieldInFilter(
+    field: FieldPath,
+    op: Operator,
+    value: api.Value
+  ): FieldFilter {
+    debugAssert(
```

Comment:
You can avoid this assertion by making this a part of the type of the parameter. I believe you can declare `op` like this:

```ts
op: Operator.IN | Operator.NOT_IN
```

and then the caller will have to be in a context where this is true.